### PR TITLE
feat: support css url import dependency

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
1. Use `Span` instead of `module_request_strart` since the span of URL is more complicated and can't use tricks to reduce memory.
2. try to resolve https://github.com/rolldown/rolldown/blob/63875e40b7687f9917b70556c23f60e925b111aa/scripts/snap-diff/stats/aggregated-reason.md#L114-L114

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
* **#2600** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2600?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2599** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2599?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2598** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2598?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2587** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2587?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @IWANABETHATGUY and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/rolldown/tests/esbuild/default/metafile_various_cases/artifacts.snap [hunk 1]
```diff
@@ -33,9 +33,34 @@ snapshot_kind: text
...
+
+```text
+[UNRESOLVED_IMPORT] Warning: Could not resolve extern.png in entry.css
+
 ```
 # Assets
 
+## assets/file-JEFwJEkD.file
+
+## copy.js
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/default/metafile_various_cases/artifacts.snap:47]: Since we will fallback to `js` module type if there is no `loader` is specify, `copy.copy` is used `js` loader to process.

File: crates/rolldown/tests/esbuild/default/metafile_no_bundle/artifacts.snap
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/default/metafile_no_bundle/artifacts.snap:1]: Those warnings are expected, they can't be found in testcase dir

File: crates/rolldown/tests/esbuild/loader/loader_file_one_source_two_different_output_paths_css/artifacts.snap [hunk 2]
```diff
@@ -4,18 +4,24 @@ snapshot_kind: text
...
-	background: url(common.png);
+	background: url(assets/common-h8VKzHVD.png);
 }
 
 ```
 ## common.js
 
 ```js
 
+//#region shared/common.png
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/loader/loader_file_one_source_two_different_output_paths_css/artifacts.snap:21]: This redundant file is not related to this pr, since we will create `ecma_view` related to asset whatever before.

**Comment by IWANABETHATGUY** [comment]
### Merge activity

* **Nov 4, 11:22 PM EST**: [Graphite](https://app.graphite.dev/github/pr/rolldown/rolldown/2598) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

File: crates/rolldown/src/stages/generate_stage/mod.rs [hunk 3]
```diff
@@ -293,13 +293,37 @@ impl<'a> GenerateStage<'a> {
...
+      chunk.modules.iter().for_each(|module_idx| {
+        let module = &mut self.link_output.module_table.modules[*module_idx];
+        if let Some(css_view) =
+          module.as_normal_mut().and_then(|normal_module| normal_module.css_view.as_mut())
+        {
+          let mut mutation_span_pairs = vec![];
+          for (idx, record) in css_view.import_records.iter_enumerated() {
+            if let Some(asset_filename) = module_idx_to_filenames.get(&record.resolved_module) {
+              let span = css_view.record_idx_to_span[idx];
+              mutation_span_pairs.push((span, asset_filename.clone()));
```
Comment (underfin) [inline crates/rolldown/src/stages/generate_stage/mod.rs]: Here could better direct using 

```
css_view
              .mutations
              .push(Box::new(CssAssetNameReplacer { span, asset_name: asset_filename }));
```

File: crates/rolldown_common/src/types/import_record.rs [hunk 4]
```diff
@@ -12,11 +13,11 @@ oxc::index::define_index_type! {
 }
 
 #[derive(Debug)]
-pub struct ImportRecordStateStart {
+pub struct ImportRecordStateSpan {
   /// Why use start_offset instead of `Span`? Cause, directly pass `Span` will increase the type
   /// size from `40` to `48`(8 bytes alignment). Since the `RawImportRecord` will be created multiple time,
   /// Using this trick could save some memory.
```
Comment (underfin) [inline crates/rolldown_common/src/types/import_record.rs]: The comment is invalid.

File: crates/rolldown/tests/esbuild/loader/loader_file_relative_path_css/artifacts.snap [hunk 5]
```diff
@@ -1,18 +1,25 @@
...
 
 ```
 ## entries_entry.js
 
 ```js
 
+//#region images/image.png
+var image_default = "assets/image-6tcw8vpN.png";
+
+//#endregion
```
Comment (underfin) [inline crates/rolldown/tests/esbuild/loader/loader_file_relative_path_css/artifacts.snap:24]: Not sure here why need to generate the js file.

Comment (IWANABETHATGUY) [reply crates/rolldown/tests/esbuild/loader/loader_file_relative_path_css/artifacts.snap:24]: https://github.com/rolldown/rolldown/pull/2598#discussion_r1828673482, as I said this, the redundant file is generated because the asset file related ecma view will codegen whatever, could be polished  in the future, But I think it is not related to this pr, since this pr only handle `url()` dependency

File: crates/rolldown_common/src/types/import_record.rs (hunk 4 above)
Comment (IWANABETHATGUY) [reply crates/rolldown_common/src/types/import_record.rs]: Nice catch

**Review by underfin — APPROVED** [review]

**Review by hyf0 — APPROVED** [review]
